compiler/rustc_hir_typeck/src/closure.rs

```diff
@@ -622,7 +622,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         debug!(?body.generator_kind);
 
         let hir_id = self.tcx.hir().local_def_id_to_hir_id(expr_def_id);
-        let bound_vars = self.tcx.late_bound_vars(hir_id);
+        let bound_vars = match decl.inputs.len() {
+            0 => List::empty(),
+            _ => self.tcx.late_bound_vars(hir_id),
+        };
```

Comment:
We should be figuring out why `late_bound_vars` is having any bound vars at all.